Sources/Parsers/StoryboardParser.swift

```diff
@@ -28,90 +28,70 @@ public final class StoryboardParser {
-
-    @objc func parser(_ parser: XMLParser, didStartElement elementName: String,
-                      namespaceURI: String?, qualifiedName qName: String?,
-                      attributes attributeDict: [String: String]) {
-
-      switch elementName {
-      case "document":
-        initialViewControllerObjectID = attributeDict["initialViewController"]
-      case "scene":
-        inScene = true
-      case "objects" where inScene:
-        readyForFirstObject = true
-      case let tag where (readyForFirstObject && tag != "viewControllerPlaceholder"):
-        let customClass = attributeDict["customClass"]
-        let customModule = attributeDict["customModule"]
-        if let objectID = attributeDict["id"], objectID == initialViewControllerObjectID {
-          initialScene = InitialScene(objectID: objectID,
-                                      tag: tag,
-                                      customClass: customClass,
-                                      customModule: customModule)
-        }
-        if let storyboardID = attributeDict["storyboardIdentifier"] {
-          scenes.insert(Scene(storyboardID: storyboardID,
-                              tag: tag,
-                              customClass: customClass,
-                              customModule: customModule))
-        }
-        readyForFirstObject = false
-      case "connections":
-        readyForConnections = true
-      case "segue" where readyForConnections:
-        if let segueID = attributeDict["identifier"] {
-          let customClass = attributeDict["customClass"]
-          let customModule = attributeDict["customModule"]
-          segues.insert(Segue(identifier: segueID, customClass: customClass, customModule: customModule))
-        }
-      default:
-        break
-      }
-    }
 
-    @objc func parser(_ parser: XMLParser, didEndElement elementName: String,
-                      namespaceURI: String?, qualifiedName qName: String?) {
-      switch elementName {
-      case "scene":
-        inScene = false
-      case "objects" where inScene:
-        readyForFirstObject = false
-      case "connections":
-        readyForConnections = false
-      default:
-        break
+    for scene in document.xpath(XMLScene.path) {
+      guard scene.tag != XMLScene.placeholderTag else { continue }
+
+      let customClass = scene[XMLScene.customClassAttribute]
+      let customModule = scene[XMLScene.customModuleAttribute]
+
+      if scene[XMLScene.idAttribute] == initialSceneID {
+        initialScene = InitialScene(tag: scene.tag ?? "",
+                                    customClass: customClass,
+                                    customModule: customModule)
+      }
+      if let id = scene[XMLScene.storyboardIdentifierAttribute] {
+        scenes.insert(Scene(storyboardID: id,
+                            tag: scene.tag ?? "",
+                            customClass: customClass,
+                            customModule: customModule))
```

Comment:
Scenes can have no `storyboardIdentifier`, in that case they're anonymous scenes. Usually such scenes are connected via segues to other scenes. Anonymous scenes cannot be directly instantiated from code, exactly because they don't have an identifier.

In all my projects, most of my scenes don't have an identifier exactly because they're all connected via segues. Only some of them have an identifier, and those correctly appear in the swiftgen generated file.